add section to environments (#674)
dataform-co/dataform: content/docs/dataform-web/guides/environments.md

  ]
}
```

> ## Example: separating development and production data into two databases

Reviewer: ```suggestion
## Example: use separate databases for development and production data
```

Some teams may not be at the stage where they require a `staging` environment, but still would like to keep development and production data separated. This can be done using a `configOverride` in the `production` environment.

In the example below:

> - any code deployed whilst developing will use the `defaultDatabase` from the `dataform.json`

Reviewer: ```suggestion
- any code deployed during development will use the `defaultDatabase` from the `dataform.json`
```

seems closer to American English :(

> - any code deployed by schedules will be in the `production` environment and so use the `defaultDatabase` from the `configOverride` in `environmemts.json`

Reviewer: ```suggestion
- any code deployed by schedules will run in the `production` environment and so use the `defaultDatabase` from the `configOverride` in `environments.json`
```

> - any code deployed by schedules will be in the `production` environment and so use the `defaultDatabase` from the `configOverride` in `environmemts.json`

Reviewer: suggest rewording these last two bullets. they make it sounds like a schedule <-> environment linkage is sort of dynamic, or that you might be able to configure it, but that's not the case. i suggest something like:

```suggestion
- schedules only run in the single `production` environment, and so use the `defaultDatabase` from that environment's `configOverride`
- the `production` environment specifies the `master` Git branch, so all of its schedules will run using that version of the code
```

- schedules will use the version of the code from the `master` branch

`dataform.json`:
```json
{
    "warehouse": "bigquery",
    "defaultSchema": "dataform_data",
    "defaultDatabase": "analytics-development",
}
```

`environments.json`:
```json
 {
    "environments": [
        {
            "name": "production",
            "gitReference": {
                "branch": "master"
            },

>             "configOverride": {"defaultDatabase": "analytics-production"}

Reviewer: ```suggestion
            "configOverride": { "defaultDatabase": "analytics-production" }
```
